### PR TITLE
Projects/EfficientDet

### PR DESCRIPTION
**Pull request by zwhus**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

BiFPN neck codes and train config

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: projects/BiFPN/BiFPN/bifpn.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
+from typing import List
```
Comment (ZwwWayne) [inline projects/BiFPN/BiFPN/bifpn.py:1]: The file structure can be 
- projects/BiFPN
    - bifpn
        - bifpn.py
        - utils.py

File: projects/BiFPN/REMEAD.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    <!-- The code's design shall follow existing interfaces and convention. For example, each model component should be registered into `mmdet.registry.MODELS` and configurable via a config file. -->
+
+  - [x] Basic docstrings & proper citation
+
+    <!-- Each major object should contain a docstring, describing its functionality and arguments. If you have adapted the code from other open-source projects, don't forget to cite the source project in docstring and make sure your behavior is not against its license. Typically, we do not accept any code snippet under GPL license. [A Short Guide to Open Source Licenses](https://medium.com/nationwide-technology/a-short-guide-to-open-source-licenses-cf5b1c329edd) -->
+
+  - [x] Test-time correctness
+
+    <!-- If you are reproducing the result from a paper, make sure your model's inference-time performance matches that in the original paper. The weights usually could be obtained by simply renaming the keys in the official pre-trained weights. This test could be skipped though, if you are able to prove the training-time correctness and check the second milestone. -->
```
Comment (ZwwWayne) [inline projects/BiFPN/REMEAD.md:80]: We can add some description about the Test-time correctness, e.g.,
```
Have checked that the same tensor can obtain the same output tensor as the official implementations (also add link for official implementation)/
```

File: projects/EfficientDet/REMEAD.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+</div>
+
+## Description
+
+This is an implementation of [EfficientDet](https://github.com/google/automl) based on [MMDetection](https://github.com/open-mmlab/mmdetection/tree/3.x), [MMCV](https://github.com/open-mmlab/mmcv), and [MMEngine](https://github.com/open-mmlab/mmengine).
+<br>
+EfficientDet a new family of object detectors, which consistently achieve much better efficiency than prior art across a wide
+spectrum of resource constraints.
+<br>
+In particular, with singlemodel and single-scale, EfficientDet-D7 achieves stateof-the-art 55.1 AP on COCO test-dev with 77M parameters and 410B FLOP
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:19]: single model -> single model

File: projects/EfficientDet/REMEAD.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+### convert weights from tensorflow to pytorch
+
+```bash
+python efficientdet/transfer_efficientdet --backbone {backname} --tensorflow_weight {tensorflow_efficientdet_path} --out_weight {out_path}
+```
+
+### For Example
+
+```bash
+python efficientdet/transfer_efficientdet --backbone efficientnet-b0 --tensorflow_weight efficientdet-d0/model --out_weight demo.pth
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:41]: - we can move this file to `projects/EfficientDet/transfer_efficientdet.py`
- better to rename it as convert_tf_to_pt.py
- suggest to indicate users they need to install tensorflow if they want to use this script.

File: projects/EfficientDet/REMEAD.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+### Testing commands
+
+In MMDetection's root directory, run the following command to test the model:
+
+```bash
+python tools/train.py projects/EfficientDet/configs/efficientdet_effb0_bifpn_8xb16-crop512-300e_coco.py ${CHECKPOINT_PATH}
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:49]: better to provide test instead of training command.

File: projects/EfficientDet/REMEAD.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+In contrast to other feature pyramid network, such as FPN, FPN + PAN, NAS-FPN, BiFPN achieves  the best accuracy with fewer parameters and FLOPs.
+
+<div align="center">
+<img src="https://github.com/zwhus/pictures/raw/main/Screenshot%20from%202023-01-30%2015-33-42.png">
+</div>
+<br>
+
+## Usage
+
+#### In MMDetection's root directory, if you want convert weights from tensorflow to pytorch, run the following command to test the model:
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:32]:
```suggestion
#### Model conversion

In MMDetection's root directory, if you want convert weights from tensorflow to pytorch, run the following command to test the model:
```

File: projects/EfficientDet/REMEAD.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+```bash
+python projects/EfficientDet/convert_tf_to_pt.py --backbone {backname} --tensorflow_weight {tensorflow_efficientdet_path} --out_weight {out_path}
+```
+
+#### For Example
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:38]:
```suggestion
#### Test
```

File: projects/EfficientDet/REMEAD.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+```bash
+python efficientdet/transfer_efficientdet --backbone efficientnet-b0 --tensorflow_weight efficientdet-d0/model --out_weight demo.pth
+```
+
+#### Note: if you want convert weights from tensorflow to pytorch, you should firstly install tensorflow, please use the following command
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:44]:
```suggestion
**Note：** if you want convert weights from tensorflow to pytorch, you should firstly install tensorflow, please use the following command
```

File: projects/EfficientDet/REMEAD.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
+# EfficientDet
+
+> [**EfficientDet: Scalable and Efficient Object Detection**](https://arxiv.org/pdf/1911.09070.pdf),
+> Mingxing Tan, Ruoming Pang, Quoc V. Le
+> *CVPR 2020*
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:5]: should also add ','

File: projects/EfficientDet/REMEAD.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+<br>
+In contrast to other feature pyramid network, such as FPN, FPN + PAN, NAS-FPN, BiFPN achieves  the best accuracy with fewer parameters and FLOPs.
+
+<div align="center">
+<img src="https://github.com/zwhus/pictures/raw/main/Screenshot%20from%202023-01-31%2010-38-51.png">
+</div>
+
+## Usage
+
+### Model conversion
```
Comment (ZwwWayne) [inline projects/EfficientDet/REMEAD.md:25]: We should also tell users where to download these pre-trained model

File: projects/EfficientDet/configs/efficientdet_effb0_bifpn_16xb8-crop512-300e_coco.py [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+    'mmdet::_base_/datasets/coco_detection.py',
+    'mmdet::_base_/schedules/schedule_1x.py',
+    'mmdet::_base_/default_runtime.py'
+]
+custom_imports = dict(
+    imports=['projects.EfficientDet.efficientdet'], allow_failed_imports=False)
+
+image_size = 512
+datasettype = 'Coco90Dataset'
+evalute_type = 'Coco90Metric'
```
Comment (RangiLyu) [inline projects/EfficientDet/configs/efficientdet_effb0_bifpn_16xb8-crop512-300e_coco.py:11]: unused parameter

File: projects/EfficientDet/configs/efficientdet_effb0_bifpn_16xb8-crop512-300e_coco_xy.py [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+    'mmdet::_base_/datasets/coco_detection.py',
+    'mmdet::_base_/schedules/schedule_1x.py',
+    'mmdet::_base_/default_runtime.py'
+]
+custom_imports = dict(
+    imports=['projects.EfficientDet.efficientdet'], allow_failed_imports=False)
+
+image_size = 512
+datasettype = 'Coco90Dataset'
+evalute_type = 'Coco90Metric'
```
Comment (RangiLyu) [inline projects/EfficientDet/configs/efficientdet_effb0_bifpn_16xb8-crop512-300e_coco_xy.py:11]: unused

**Review by ZwwWayne — APPROVED** [review]
